### ISSUE TITLE
Configure errors in perf testing build

### ISSUE THREAD

**Issue by brian-kelley** [kokkos/kokkos#3632]
When building Kokkos as a Trilinos package (Trilinos develop branch, so 3.2.1), when I set ``-DTrilinos_TEST_CATEGORIES=PERFORMANCE`` to get a performance testing build, these errors come up during configure:

```
CMake Error at packages/kokkos/cmake/fake_tribits.cmake:114 (add_dependencies):
  Cannot add target-level dependencies to non-existent target
  "KokkosCore_ProfilingAllCalls".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.
Call Stack (most recent call first):
  packages/kokkos/core/unit_test/CMakeLists.txt:438 (KOKKOS_ADD_TEST)


CMake Error at packages/kokkos/core/unit_test/CMakeLists.txt:521 (TARGET_INCLUDE_DIRECTORIES):
  Cannot specify include directories for target
  "KokkosCore_IncrementalTest_OPENMP" which is not built by this project.
Call Stack (most recent call first):
  packages/kokkos/core/unit_test/CMakeLists.txt:526 (KOKKOS_ADD_INCREMENTAL_TEST)


CMake Error at packages/kokkos/core/unit_test/CMakeLists.txt:521 (TARGET_INCLUDE_DIRECTORIES):
  Cannot specify include directories for target
  "KokkosCore_IncrementalTest_SERIAL" which is not built by this project.
Call Stack (most recent call first):
  packages/kokkos/core/unit_test/CMakeLists.txt:526 (KOKKOS_ADD_INCREMENTAL_TEST)
```

The last two errors do seem to be affected by the backends I enable - for each enabled backend, there's an error about ``KokkosCore_IncrementalTest_$BACKEND``.

**Comment by crtrott** [comment]
is this fixed by something in 3.3?

**Comment by brian-kelley** [comment]
@crtrott I dropped in develop branch Kokkos and added ``-DCMAKE_CXX_STANDARD=14``, but got the same errors.

**Comment by brian-kelley** [comment]
Also tried branch ``release-candidate-3.3.0``

**Comment by brian-kelley** [comment]
#3634 has a fix.

**Comment by ndellingwood** [comment]
Merged with part of 3.2.1 release
